api/comments/serializers.py

```diff
@@ -43,8 +43,8 @@ class CommentSerializer(JSONAPISerializer):
 
     target = TargetField(link_type='related', meta={'type': 'get_target_type'})
     user = RelationshipField(related_view='users:user-detail', related_view_kwargs={'user_id': '<user._id>'})
-    node = RelationshipField(related_view='nodes:node-detail', related_view_kwargs={'node_id': '<node._id>'})
-    replies = RelationshipField(self_view='nodes:node-comments', self_view_kwargs={'node_id': '<node._id>'}, filter={'target': '<pk>'})
+    node = RelationshipField(related_view='registrations:registration-detail', related_view_kwargs={'node_id': '<node._id>'})
```

Comment:
And looking here, I'm pretty sure that you're just not handling the case where it's a node. This is a tricky one, so don't be surprised if it takes another day or two to fix.